Libraries/LibTLS/ClientHandshake.cpp

```diff
@@ -363,6 +363,29 @@ ssize_t TLSv12::handle_payload(const ByteBuffer& vbuffer)
                     ASSERT_NOT_REACHED();
                 }
                 payload_res = handle_certificate(buffer.slice_view(1, payload_size));
+                if (m_context.certificates.size()) {
+                    auto one_valid { false };
+                    size_t valid_index { 0 };
+                    for (auto& cert : m_context.certificates) {
+                        if (cert.is_valid()) {
+                            one_valid = true;
+                            break;
+                        }
+                        ++valid_index;
+                    }
```

Comment:
How about using `Vector::find` here?

```cpp
auto it = m_context.certificates.find([&](auto& cert) { return cert.is_valid(); });
...
```

Maybe it's made more awkward by the iterator interface after that.. What do you think?

Reply:
I'll try.

Reply:
The iterator interface is just fine (`it.index()` exists!)